packages/core/src/createProps.ts

```diff
@@ -238,7 +238,7 @@ const getPropTableObject = () => {
     return tableObject
 }
 
-export const createProps = (overrides?: Partial<CCSSProps>): Partial<CCSSProps> => {
+export const createProps = <T>(overrides?: Partial<T | CCSSProps>): Partial<T | CCSSProps> => {
```

Comment:
Try `extends`

Reply:
This should be enough

```
export const createCCSS = (options: Partial<CCSSOptions> = defaultOptions): CCSS => {
    const __ccss = ccssProps => generate(ccssProps, options)
    options.__ccss = __ccss
    return __ccss
}
```

Reply:
This is enough to solve the error on `__ccss`. The issue is that when we want to add a different property to extend CCSS props, we get an error where the custom prop has no properties in common with type Partial<CCSSProps> : (

Example:
```
const props = createProps({
    size: pipe(mapValue, (input, prop, options) => options.__ccss(input))
})

/*
error TS2345: Argument of type '{ props: { size: CCSSParser; }; valueMap: any; }' is not assignable to parameter of type 'Partial<CCSSOptions>'.
  Types of property 'props' are incompatible.
    Type '{ size: CCSSParser; }' has no properties in common with type 'Partial<CCSSProps>'.
*/
const ccss = createCCSS(createOptions({ props, valueMap}))
```

Reply:
@wintercounter I solved the fucking issue.
I created a type intersection between them (that's the `&` operator). Done. Suck it TS!!